ml_experiment/metadata/MetadataTableRegistry.py

```diff
@@ -0,0 +1,112 @@
+import sqlite3
+from ml_experiment.DefinitionPart import ValueType
+import ml_experiment._utils.sqlite as sqlu
+
+from typing import Dict, Iterable
+from ml_experiment.metadata.MetadataTable import MetadataTable
+
+
+class MetadataTableRegistry:
```

Comment:
The design principle for these two classes (`MetadataTableRegistry` and `MetadataTable`) is to avoid storing any observable state information. In this way, each of these classes could be equivalently rewritten as collections of pure functions instead of classes.

I used classes in order to have some transparent caching associated with certain queries. This way, the abstraction doesn't cause us to lose too much performance due to (accidentally) repeated queries. Trying to strike the balance between performance and readability here.